src/cloudcasting_metrics/app.py

```diff
@@ -0,0 +1,190 @@
+
+# ---------------------------------------------------------------------------
+
+# The forecast produces these horizon steps
+FORECAST_STEPS = pd.timedelta_range(start="15min", end="180min", freq="15min")
+# The forecast is run at this frequency
+FORECAST_FREQ = pd.Timedelta("30min")
+
+
+def open_icechunk(path: str) -> xr.Dataset:
+    """Open an icechunk store to xarray Dataset
+    
+    Args:
+        path: The path to the local or s3 icechunk store
+    """
+
+    if path.startswith("s3://"):
+        bucket, _, path = path.removeprefix("s3://").partition("/")
+        store = icechunk.s3_storage(
+            bucket=bucket,
+            prefix=path,
+            access_key_id=os.environ["AWS_ACCESS_KEY_ID"],
+            secret_access_key=os.environ["AWS_SECRET_ACCESS_KEY"],
+            region=os.environ["AWS_REGION"],
+        )
+    else:
+        store = icechunk.local_filesystem_storage(path=path)
+
+    repo = icechunk.Repository.open(store)
+    session = repo.readonly_session("main")
+    return xr.open_zarr(session.store)
+
+
+def app(date: pd.Timestamp | None = None) -> None:
+    """Runs metric calculations on cloudcasting for a given input day and appends to zarr store
+
+    Args:
+        date: The day for which the cloudcasting predictions will be scored.
+    """
+
+    # Unpack environmental variables
+    sat_path = os.environ["SATELLITE_ICECHUNK_ARCHIVE"]
+    prediction_dir = os.environ["PREDICTION_SAVE_DIRECTORY"]
+    metric_zarr_path = os.environ["METRIC_ZARR_PATH"]
+
+
+    now = pd.Timestamp.now(tz="UTC").replace(tzinfo=None)
+
+    # Default to yesterday
+    if date is None:
+        date = now.floor("1D") - pd.Timedelta("1D")
+    
+    start_dt =  date.floor("1D")
+    end_dt = date.floor("1D") + pd.Timedelta("1D")
+
+    if now <= end_dt + FORECAST_STEPS.max():
+        raise Exception(
+            f"We cannot score forecast with init-time {end_dt} until after the last valid-time."
+        )
+
+    # Open the satellite data store
+    ds_sat = open_icechunk(path=sat_path)
+
+    # Slice to only the timesteps we need for scoring
+    ds_sat = ds_sat.sel(time=slice(start_dt, end_dt + FORECAST_STEPS.max()))
+
+    # It is better to preload if we have the RAM space
+    # - This eliminates any costs of repeatedly streaming data from the bucket
+    # - It's also faster
+    ds_sat = ds_sat.compute()
+
+    # Find recent forecasts
+    date_string = start_dt.strftime("%Y-%m-%d")
+    remote_path = f"{prediction_dir}/{date_string}*.zarr"
+    fs, path = fsspec.core.url_to_fs(remote_path)
+
+    file_list = fs.glob(path)
+
+    # Filter forecasts
+    # - We only score forecasts we have the satellite data for
+    # - If we are missing one satellite image we will skip scoring all forecasts require that
+    forecasts_to_score = []
+
+    for file in file_list:
+        # Find the datetime of this forecast
+        match = re.search(r'\d{4}-\d{2}-\d{2}T\d{2}:\d{2}', file)
+        assert match
```

Comment:
i think its slightly better to do
if match is None:
   raise Exception("Tried lookinf for .... in file {file}, but oculdnt find it")